pkgs/by-name/un/unblob/package.nix

```
@@ -0,0 +1,107 @@
+{
+  lib,
+  callPackage,
+  python3,
+  fetchFromGitHub,
+  gitUpdater,
+  makeWrapper,
+  e2fsprogs,
+  jefferson,
+  lz4,
+  lziprecover,
+  lzop,
+  p7zip,
+  sasquatch,
+  sasquatch-v4be,
+  simg2img,
+  ubi_reader,
+  unar,
+  zstd,
+}:
+
+let
+  # These dependencies are only added to PATH
+  runtimeDeps = [
+    e2fsprogs
+    jefferson
+    lziprecover
+    lzop
+    p7zip
+    sasquatch
+    sasquatch-v4be
+    ubi_reader
+    simg2img
+    unar
+    zstd
+    lz4
+  ];
+in
+python3.pkgs.buildPythonApplication rec {
+  pname = "unblob";
+  version = "24.9.30";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "onekey-sec";
+    repo = "unblob";
+    rev = version;
+    hash = "sha256-r5UGoCOzYO/v7cAkbhOmeX7caspues/k/gokuzh1wsU=";
+    forceFetchGit = true;
+    fetchLFS = true;
+  };
+
+  strictDeps = true;
+  buildInputs = with python3.pkgs; [ poetry-core ];
+
+  propagatedBuildInputs = with python3.pkgs; [
```

```suggestion
  build-system = with python3.pkgs; [ poetry-core ];

  dependencies = with python3.pkgs; [
```
this should be used with pyproject = true